developer_tests/run_tests.csh

```diff
@@ -142,16 +142,21 @@ foreach TESTFILE ( $HAS_TESTS )
 
            set FAILURE = 0
            set PROG = `echo $TARGET | sed -e 's#mkmf_##'`
+           set THISLOG = ${LOGDIR}/runlog.${LOGNAME}.${PROG}.out 
            echo Starting $PROG
 
-           if ( -f using_mpi_for_$PROG ) then
-              ( ${MPICMD} ./$PROG >> ${LOGDIR}/runlog.${LOGNAME}.${PROG}.out ) || set FAILURE = 1
+           if ( -f using_mpi_for_$PROG && -f ${PROG}.in ) then
+              ( ${MPICMD} ./$PROG < ${PROG}.in >> $THISLOG ) || set FAILURE = 1
+           else if ( -f using_mpi_for_$PROG ) then
+              ( ${MPICMD} ./$PROG              >> $THISLOG ) || set FAILURE = 1
+           else if ( -f ${PROG}.in ) then
+              (           ./$PROG < ${PROG}.in >> $THISLOG ) || set FAILURE = 1
            else
-              (           ./$PROG >> ${LOGDIR}/runlog.${LOGNAME}.${PROG}.out ) || set FAILURE = 1
+              (           ./$PROG              >> $THISLOG ) || set FAILURE = 1
            endif
          
            if ( $FAILURE ) then
-              echo "ERROR - unsuccessful run of $PROG"
+              echo "ERROR - unsuccessful run of $PROG at "`date`
```

Comment:
```suggestion
              echo "ERROR - unsuccessful run of $PROG"
```

Reply:
I find that the addition of the date to failed/unsuccessful situations to be useful when comparing the run-time output of two tests.  If they run OK - there is no need to datestamp them. When 'diffing' two files, your attention is not drawn to successful results. If they fail ... the difference in date is enough to make them stand out.